src/AlgebraicGeometry/TropicalGeometry/groebner_fan.jl

```diff
@@ -0,0 +1,544 @@
+
+    return weight_ordering(Int.(homogeneityVector),
+                           weight_ordering(Int.(interior_facet_point),
+                                           weight_ordering(Int.(outer_normal_vector),
+                                                           revlex(R))))
+
+end
+function groebner_flip_adjacent_ordering(R::MPolyRing,
+                                         homogeneityVector::Nothing,
+                                         interior_facet_point::Vector{fmpz},
+                                         outer_normal_vector::Vector{fmpz})
+
+    return weight_ordering(Int.(interior_facet_point),
+                           weight_ordering(Int.(outer_normal_vector),
+                                           revlex(R)))
+
+end
+
+
+
+# constructs a polyhedral fan from list of cones
+function polyhedral_fan_from_cones(listOfCones::Vector{<:Cone})
+
+    ###
+    # Collect incidence information
+    ###
+    fanRays = Vector{Vector{fmpq}}()
+    fanIncidences = Vector{Vector{Int}}()
+    for cone in listOfCones
+        coneIncidence = Vector{Int}()
+        coneRays,_ = rays_modulo_lineality(cone)
+        for coneRay in coneRays
+            # if ray does not occur in rayList, add it to it
+            # either way add the corresponding index to rayIncidences
+            i = findfirst(isequal(coneRay),fanRays)
+            if i === nothing
+                push!(fanRays,coneRay)
+                push!(coneIncidence,length(fanRays))
+            else
+                push!(coneIncidence,i)
+            end
+        end
+        push!(fanIncidences,coneIncidence)
+    end
+
+    # convert Vector{Vector} to IncidenceMatrix and Matrix respectively
+    fanIncidences = IncidenceMatrix(fanIncidences)
+    fanRays = matrix(QQ,fanRays)
+
+    fanLineality = matrix(QQ,lineality_space(first(listOfCones)))
+
+    return PolyhedralFan(fanRays,fanLineality,fanIncidences)
+end
+
+
+
+
+@doc raw"""
+    groebner_fan(I::MPolyIdeal; return_groebner_bases::Bool=false, return_orderings::Bool=false, verbose_level::Int=0)
+
+# Input
+- `I`: A multivariate polynomial ideal over a coefficient field
+- `return_groebner_bases`: If `true`, return all Groebner bases computed during the traversal
+- `return_orderings`: If `true`, return all monomial orderings used during the traversal
+- `verbose_level`: If positive, print out how many cones have been computed during the traversal
+
+# Output
+A `PolyhedralFan` that is the Groebner fan of `I`.
+
+If `I` is not weighted homogeneous with respect to a positive weight vector, the Groebner fan will be restricted to the positive orthant.  Otherwise, the Groebner fan will span the entire space and have a non-trivial lineality space.
+
+If `return_groebner_bases==true`, returns a dictionary whose keys are interior points of the maximal cones and whose values are the Groebner bases for those cones.  Their union will be a universal Groebner basis.
```

Comment:
My understanding is that the dictionary is returned *in addition* to the fan, is that right? I'd suggest to make this clear, one way or the other. E.g.:
```suggestion
If `return_groebner_bases==true`, also return a dictionary whose keys are interior points of the maximal cones and whose values are the Groebner bases for those cones.  Their union will be a universal Groebner basis.
```